python/lsst/daf/butler/registries/oracleRegistry.py

```diff
@@ -107,6 +120,28 @@ def _oracleExecute(connection, cursor, statement, parameters, context, executema
                     # single quote, optional close double quote
                     statement = re.sub(f"\"?'?{name}\\b'?\"?", lambda x: _ignoreQuote(name, x), statement)
             return statement, parameters
-        engine = create_engine(self.config["db"], pool_size=1)
+
+        def _connCreator(**kw):
+            """ This function sets some Oracle values for each new pool
+            connection (e.g., module, schema).  Add connect listener to
+            Pool even if pool_size=1 for automatic reconnects.
+            """
+            # get executable name to save as DBMS_APPLICATION_INFO module
+            try:
+                modname = __main__.__file__
+            except AttributeError:
+                modname = "Unknown"
+            modname = os.path.basename(modname)[:48]  # column limited to 48 chars
+
+            # get the connection
+            dbapi = kw['dbapi_connection']
+
+            # cx_Oracle specific means to set current schema and module
+            dbapi.module = modname
+            if ".db.schema" in self.config:
+                dbapi.current_schema = self.config[".db.schema"]
+        event.listen(Pool, "connect", _connCreator, named=True)
```

Comment:
I was thinking about this, and the reason we needed this instead of just using the one in metadata is because of the views creation right? That is already special cased for each dialect. Looking at that code a bit, if we add metadata that is already passed into the class to the CreateView class here: https://github.com/lsst/daf_butler/blob/master/python/lsst/daf/butler/core/views.py#L109, then in the oracle sql string we can format it before executing it to supply the schema from the metadata in front of the table names, i.e. f"...{element.metadata.schema}.user_views  ..." .  This should remove the need for this block of code.